### PR TITLE
cache is not global now

### PR DESCRIPTION
**Pull request by simon300000**
My English is not good so I didn’t change README

```javascript
const cache = require('memory-cache');
const cache2 = require('memory-cache');

cache.put('foo', '233');
cache2.put('foo', '666');

console.log(cache.get('foo'))//233
console.log(cache2.get('foo'))//666
```

And global lifetime

```javascript
const cache = require('memory-cache');
const cache2 = require('memory-cache');
cache.life(1000); // Time in ms

cache.put('foo', 'bar');
cache2.put('foo', 'alive')
console.log(cache.get('foo')); //bar


cache.put('houdini', 'disappear', 2000);
console.log('Houdini will ' + cache.get('houdini')); //Houdini will disappear

setTimeout(function() {
    console.log('Houdini is not ' + cache.get('houdini')); //Houdini is not disappear
    console.log('foo is ' + cache.get('foo')); //foo is null
    console.log('I\'m still ' + cache2.get('foo'));//I'm still alive
}, 1500);

setTimeout(function() {
    console.log('Houdini is ' + cache.get('houdini')); //Houdini is null
}, 2500);
```

Able to set max size, drop base on the last hit
```javascript
cache.maxSize(3);

cache.put('a', 'a');
cache.put('b', 'b');
cache.put('c', 'c');
cache.get('a');
cache.put('d', 'd');
cache.put('e', 'e');

console.log(cache.get('a')); //a
console.log(cache.get('b')); //null
console.log(cache.get('c')); //null
console.log(cache.get('d')); //d
console.log(cache.get('e')); //e
```

#55 #43

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+'use strict';
```
Comment (ptarjan) [inline app.js]: What is the point of this file?

File: gulpfile.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ gulp.task('test', function() {
   return gulp.src(paths.js)
     .pipe(istanbul())
     .pipe(istanbul.hookRequire())
-    .on('finish', function () {
+    .on('finish', function() {
```
Comment (ptarjan) [inline gulpfile.js]: This change is unrelated. If you want to add some linting for this, i'm game, but do it in another PR please.

File: index.js [hunk 3]
```diff
@@ -99,7 +114,27 @@ exports.get = function(key) {
 };
 
 exports.size = function() {
-  return size;
+  return lastHit.length;
+};
+
+exports.life = function(value) {
```
Comment (ptarjan) [inline index.js]: Don't add a new API please.

**Comment by ptarjan** [comment]
This totally breaks the existing API. If you want to make a non-global cache option you can, but the default still has to be a global cache.

**Comment by simon300000** [comment]
OK
I'll leave it myself

File: gulpfile.js (hunk 2 above)
Comment (simon300000) [inline gulpfile.js]: I guess this is the problem from auto format

File: app.js (hunk 1 above)
Comment (simon300000) [inline app.js]: delete cache so different require will not return same thing

File: gulpfile.js (hunk 2 above)
Comment (simon300000) [inline gulpfile.js]: I got problem of Github's pull request
I send the pull request out, but when I commit something new in my fork, it auto goes to this PR
